src/Splat/ServiceLocation/Locator.cs

```diff
@@ -115,5 +123,27 @@ public static IDisposable RegisterResolverCallbackChanged(Action callback)
                 }
             });
         }
+
+        /// <summary>
+        /// This method will prevent resolver changed notifications from happening until
+        /// the returned <see cref="IDisposable"/> is disposed.
+        /// </summary>
+        /// <returns>A disposable which when disposed will indicate the change
+        /// notification is no longer needed.</returns>
+        public static IDisposable SuppressResolverCallbackChangedNotifications()
+        {
+            _resolverChangedNotificationSuspendCount++;
+
+            return new ActionDisposable(() => _resolverChangedNotificationSuspendCount--);
```

Comment:
Is the thought here that the disposable will keep the count at zero?  So our expectation is users will do their registration inside a `using` statement that temporarily disables them for the lifecycle of the disposable?  I guess this makes sense because a permanent disable could be bad.

Reply:
Basically. Very similar code to what we do in ReactiveObject on the rxui side or the rxlist. The volatile is likely not needed since int are atomic read writes as defined  by  the don't net  setandards but doesn't hurt to give the compiler a extra hint.